pkg/accounting/tally/tally.go

```diff
@@ -56,122 +52,80 @@ func (t *tally) Run(ctx context.Context) (err error) {
 			var item storage.ListItem
-			lim := t.limit
-			if lim <= 0 || lim > storage.LookupLimit {
-				lim = storage.LookupLimit
-			}
-			for ; lim > 0 && it.Next(&item); lim-- {
+			for it.Next(&item) {
 				pointer := &pb.Pointer{}
 				err = proto.Unmarshal(item.Value, pointer)
 				if err != nil {
 					return Error.Wrap(err)
 				}
-				pieces := pointer.Remote.RemotePieces
-				var nodeIDs storj.NodeIDList
-				for _, p := range pieces {
-					nodeIDs = append(nodeIDs, p.NodeId)
+				remote := pointer.GetRemote()
+				if remote == nil {
+					t.logger.Debug("no remote segment on pointer")
+					continue
 				}
-				online, err := t.onlineNodes(ctx, nodeIDs)
-				if err != nil {
-					return Error.Wrap(err)
+				pieces := remote.GetRemotePieces()
+				if pieces == nil {
+					t.logger.Debug("no pieces on remote segment")
+					continue
+				}
+				segmentSize := pointer.GetSegmentSize()
+				redundancy := remote.GetRedundancy()
+				if redundancy == nil {
+					t.logger.Debug("no redundancy scheme present")
+					continue
 				}
-				go t.tallyAtRestStorage(ctx, pointer, online, client)
+				minReq := redundancy.GetMinReq()
+				if minReq <= 0 {
+					t.logger.Debug("pointer minReq must be an int greater than 0")
+					continue
+				}
+				pieceSize := segmentSize / int64(minReq)
+				for _, piece := range pieces {
+                    nodeData[piece.NodeId] += pieceSize
+                }
 			}
 			return nil
 		},
 	)
-	return err
-}
-
-func (t *tally) onlineNodes(ctx context.Context, nodeIDs storj.NodeIDList) (online []*pb.Node, err error) {
-	responses, err := t.overlay.BulkLookup(ctx, pb.NodeIDsToLookupRequests(nodeIDs))
-	if err != nil {
-		return []*pb.Node{}, err
-	}
-	nodes := pb.LookupResponsesToNodes(responses)
-	for _, n := range nodes {
-		if n != nil {
-			online = append(online, n)
-		}
-	}
-	return online, nil
-}
-
-func (t *tally) tallyAtRestStorage(ctx context.Context, pointer *pb.Pointer, nodes []*pb.Node, client node.Client) {
-	segmentSize := pointer.GetSegmentSize()
-	minReq := pointer.Remote.Redundancy.GetMinReq()
-	if minReq <= 0 {
-		zap.L().Error("minReq must be an int greater than 0")
-		return
-	}
-	pieceSize := segmentSize / int64(minReq)
-	for _, n := range nodes {
-		nodeAvail := true
-		var err error
-		ok := t.needToContact(n.Id)
-		if ok {
-			nodeAvail, err = client.Ping(ctx, *n)
-			if err != nil {
-				zap.L().Error("ping failed")
-				continue
-			}
-		}
-		if nodeAvail {
-			err := t.updateGranularTable(n.Id, pieceSize)
-			if err != nil {
-				zap.L().Error("update failed")
-			}
-		}
-	}
-}
-
-func (t *tally) needToContact(id storj.NodeID) bool {
-	//TODO
-	//check db if node was updated within the last time period
-	return true
+	return t.updateRawTable(ctx, nodeData)
 }
 
-func (t *tally) updateGranularTable(id storj.NodeID, pieceSize int64) error {
+func (t *tally) updateRawTable(ctx context.Context, nodeData map[storj.NodeID]int64) error {
+	t.logger.Debug("entering updateRawTable")
```

Comment:
I'm not quite following why we are logging this?

Reply:
same here